addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerListScreen.js

```diff
@@ -173,13 +171,25 @@ odoo.define('point_of_sale.PartnerListScreen', function(require) {
             let result = await this.getNewPartners();
             this.env.pos.addPartners(result);
             this.render(true);
+            if (this.state.previousQuery == this.state.query) {
+                this.state.currentOffset += result.length;
+            } else {
+                this.state.previousQuery = this.state.query;
+                this.state.currentOffset = result.length;
+            }
             return result;
         }
         async getNewPartners() {
             let domain = [];
+            const limit = 30;
+            let offset = this.state.currentOffset;
             if(this.state.query) {
                 domain = ['|', ["name", "ilike", this.state.query + "%"],
                                ["parent_name", "ilike", this.state.query + "%"]];
+
+                if (this.state.previousQuery != this.state.query) {
+                    offset = 0;
+                }
```

Comment:
history since we talked on discord: maybe reset the `currentOffset` in the searchPartner